### PR TITLE
overhaul static data with interface and new script project

### PR DESCRIPTION
**Pull request by BlakeSLC**

### CODE REVIEW

File: Common/StaticData/IStaticData.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
+﻿namespace Common.StaticData
+{
+    using Common.TableClasses;
+    using System;
+    using System.Collections.Generic;
```
Comment (carlosmathe22) [inline Common/StaticData/IStaticData.cs:5]: organize usings

File: Common/StaticData/MCM_StaticData.cs [hunk 2]
```diff
@@ -0,0 +1,132 @@
+﻿namespace Common.StaticData
+{
+    using Common.TableClasses;
```
Comment (carlosmathe22) [inline Common/StaticData/MCM_StaticData.cs:3]: organize usings in solution

File: Common/StaticData/MCM_StaticData.cs [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+        public AllChannelsProfileIds AllChannelsProfileIds => new AllChannelsProfileIds(profileID: 8000, channelIdPid: 8001, channelTitleIdx: 9);
+
+        public AllLayouts AllLayouts => new AllLayouts(allLayoutsTableId: 10300, position_Idx: 5, channelTitle_Idx: 2, layoutName_Pid: 10305, titleColumnPid: 10353);
+
+        public Outputs Outputs => new Outputs(outputsTableId: 1500, outputs_LayoutsColumnId: 1612, outputsTable_OutputColumnId: 1501);
+
+        public IReadOnlyDictionary<string, string> ChannelConfigRecordingDict => throw new NotImplementedException();
+
+        public IReadOnlyList<int> ChannelsTableIds => throw new NotImplementedException();
```
Comment (carlosmathe22) [inline Common/StaticData/MCM_StaticData.cs:130]: to clean up sonar remark, should we just initialize an empty dict?  
public IReadOnlyDictionary<string, string> ChannelConfigSeverityDict => new Dictionary<string,string>();

File: Common/StaticData/MCS_StaticData.cs [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+                {"13","60 fps"},
+            };
+
+        public AllChannelsProfileIds AllChannelsProfileIds => new AllChannelsProfileIds(profileID: 2400, channelIdPid: 2403, channelTitleIdx: 3);
+
+        public AllLayouts AllLayouts => new AllLayouts(allLayoutsTableId: 5600, position_Idx: 5, channelTitle_Idx: 2, layoutName_Pid: 5605, titleColumnPid: 5653);
+
+        public Outputs Outputs => new Outputs(outputsTableId: 3400, outputs_LayoutsColumnId: 3456, outputsTable_OutputColumnId: 3403);
+
+        public IReadOnlyDictionary<string, string> ChannelConfigSeverityDict => throw new NotImplementedException();
```
Comment (carlosmathe22) [inline Common/StaticData/MCS_StaticData.cs:130]: to clean up sonar remark, should we just initialize an empty dict?
public IReadOnlyDictionary<string, string> ChannelConfigSeverityDict => new Dictionary<string,string>();

File: Common/TableClasses/AllLayouts.cs [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+        public int LayoutName_Pid { get; }
+
+        public int Position_Idx { get; }
+
+        public int ChannelTitle_Idx { get; }
+
+        public int TitleColumnPid { get; }
+
+        public AllLayouts(int allLayoutsTableId, int layoutName_Pid, int position_Idx, int channelTitle_Idx, int titleColumnPid)
```
Comment (carlosmathe22) [inline Common/TableClasses/AllLayouts.cs:19]: white spaces and constructor should not follow a property

File: Common/TableClasses/Outputs.cs [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+    public class Outputs
+    {
+        public int TableId { get; }
+
+        public int LayoutsColumnId { get; }
+
+        public int OutputColumnId { get; }
+
+        public Outputs(int outputsTableId, int outputs_LayoutsColumnId, int outputsTable_OutputColumnId)
```
Comment (carlosmathe22) [inline Common/TableClasses/Outputs.cs:15]: constructor shouldnt follow properties

File: TAG-GQI-Infrastructure_1/TAG-GQI-Infrastructure_1.cs [hunk 7]
```diff
@@ -166,11 +166,11 @@ public GQIPage GetNextPage(GetNextPageInputArgs args)
...
-        private void GetMCMRows(List<GQIRow> rows, LiteElementInfoEvent response)
+        private void GetMCMRows(List<GQIRow> rows, LiteElementInfoEvent response, IStaticData staticInfo)
         {
             var devicesRows = SharedMethods.GetTable(_dms, response, (int)MCMTableId.DeviceOverview);
-            var cpu = Convert.ToDouble(SharedMethods.GetParameter(_dms, response, MCM.CPU_Pid));
-            var memory = Convert.ToDouble(SharedMethods.GetParameter(_dms, response, MCM.Memory_Pid));
+            var cpu = Convert.ToDouble(SharedMethods.GetParameter(_dms, response, staticInfo.CPU_Pid));
+            var memory = Convert.ToDouble(SharedMethods.GetParameter(_dms, response, staticInfo.Memory_Pid));
 
             var cloudLicenseRow = devicesRows.FirstOrDefault(x => Convert.ToString(x[0]) == "Cloud License");
```
Comment (carlosmathe22) [inline TAG-GQI-Infrastructure_1/TAG-GQI-Infrastructure_1.cs:176]: remove commented code, 185

File: TAG-GQI-Retrieve Outputs_1/TAG-GQI-Retrieve Outputs_1.cs [hunk 8]
```diff
@@ -206,7 +150,8 @@ public GQIPage GetNextPage(GetNextPageInputArgs args)
                     foreach (var response in mcmResponses.Select(x => (LiteElementInfoEvent)x))
                     {
                         var encoderConfigTable = SharedMethods.GetTable(_dms, response, (int)McmTableId.DeviceEncoderConfig);
-                        GetOutputMcmTableRows(rows, response, encoderConfigTable);
+                        var mcmStaticData = new MCM_StaticData();
```
Comment (carlosmathe22) [inline TAG-GQI-Retrieve Outputs_1/TAG-GQI-Retrieve Outputs_1.cs:153]: mcmStaticData should be declared before foreach.

File: TAG-IAS-Change Mosaic Audio_1/TAG-IAS-Change Mosaic Audio_1.cs [hunk 9]
```diff
@@ -51,12 +51,10 @@ Ambachtenstraat 33
...
 namespace TAG_IAS_Change_Mosaic_Audio_1
 {
-    using System;
-    using System.Collections.Generic;
-    using Newtonsoft.Json;
+    using SharedMethods;
     using Skyline.DataMiner.Automation;
     using Skyline.DataMiner.Utils.InteractiveAutomationScript;
-    using SharedMethods;
+    using System;
```
Comment (carlosmathe22) [inline TAG-IAS-Change Mosaic Audio_1/TAG-IAS-Change Mosaic Audio_1.cs:57]: reorganize usings

**Review by carlosmathe22 — CHANGES_REQUESTED** [review]
few remarks, mostly organize usings

**Comment by BlakeSLC** [comment]
> few remarks, mostly organize usings

Remarks fixed in latest push. small update to infrastructure script

**Pull request merged**